benchflow-experiment-manager/application/src/main/java/cloud/benchflow/experimentmanager/scheduler/ExperimentTaskScheduler.java

```diff
@@ -349,15 +381,33 @@ public void abortExperiment(String experimentID) {
           // remove from ready queue
           readyQueue.remove(experimentID);
 
+          // set test to terminated
+          experimentModelDAO.setExperimentState(experimentID, TERMINATED);
+          experimentModelDAO.setTerminatedState(experimentID, TerminatedState.ABORTED);
+
           // cancel the current running task, but let it complete before
           cancelTask(experimentID);
 
           break;
 
         case RUNNING:
 
-          // cancel the current running task, but let it complete before
-          cancelTask(experimentID);
+          RunningState runningState = experimentModelDAO.getRunningState(experimentID);
+
+          // if experiment is not already terminating
+          if (runningState != RunningState.TERMINATING) {
+
+            experimentModelDAO.setRunningState(experimentID, RunningState.TERMINATING);
+
+            // cancel the current running task, but let it complete before
+            cancelTask(experimentID);
+
+            // IF not waiting for Faban Manager we go into TERMINATING
+            if (runningState != RunningState.HANDLE_TRIAL_RESULT) {
+              handleRunningExperiment(experimentID);
```

Comment:
We need to do this because we need to run the `TERMINATING` handler in case it is not triggered by the `FabanManger` sending the result of the trial.